subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -277,6 +460,168 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 		return BT_HCI_ERR_INVALID_PARAM;
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK)
+	if ((op == BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG) ||
+	    (op == BT_HCI_LE_EXT_ADV_OP_LAST_FRAG) ||
+	    ad_len_overflow) {
+		struct pdu_adv_com_ext_adv *com_hdr_chain;
+		struct pdu_adv_com_ext_adv *com_hdr;
+		struct pdu_adv_ext_hdr *hdr_chain;
+		struct pdu_adv_aux_ptr *aux_ptr;
+		struct pdu_adv *pdu_chain_prev;
+		struct pdu_adv_ext_hdr *hdr;
+		struct pdu_adv *pdu_chain;
+		uint8_t *dptr_chain;
+		uint32_t offs_us;
+		uint16_t ter_len;
+		uint8_t *dptr;
+
+		/* Get reference to aux ptr in superior PDU */
+		(void)memcpy(&aux_ptr,
+			     &hdr_data[ULL_ADV_HDR_DATA_AUX_PTR_PTR_OFFSET],
+			     sizeof(aux_ptr));
+
+		/* Get reference to flags in superior PDU */
+		com_hdr = &pdu->adv_ext_ind;
+		hdr = (void *)&com_hdr->ext_hdr_adv_data[0];
+		dptr = (void *)hdr;
+
+		/* Allocate new PDU */
+		pdu_chain = lll_adv_pdu_alloc_pdu_adv();
+		LL_ASSERT(pdu_chain);
+
+		/* Populate the appended chain PDU */
+		pdu_chain->type = PDU_ADV_TYPE_AUX_CHAIN_IND;
+		pdu_chain->rfu = 0U;
+		pdu_chain->chan_sel = 0U;
+		pdu_chain->tx_addr = 0U;
+		pdu_chain->rx_addr = 0U;
+		pdu_chain->len = 0U;
+
+		com_hdr_chain = &pdu_chain->adv_ext_ind;
+		hdr_chain = (void *)&com_hdr_chain->ext_hdr_adv_data[0];
+		dptr_chain = (void *)hdr_chain;
+
+		/* Flags */
+		*dptr_chain = 0U;
+
+		/* TODO: should CTEInfo be added to chain PDU due to AD data
+		 *       overflow?
+		 */
+
+		/* ADI flag, mandatory if superior PDU has it */
+		if (hdr->adi) {
+			hdr_chain->adi = 1U;
+		}
+
+		/* Proceed to next byte if any flags present */
+		if (*dptr) {
+			dptr++;
+		}
+		if (*dptr_chain) {
+			dptr_chain++;
+		}
+
+		/* Start adding fields corresponding to flags here, if any */
+
+		/* No AdvA flag */
+		/* No TgtA flag */
+
+		/* CTEInfo flag */
+		if (hdr_chain->cte_info) {
+			(void)memcpy(dptr_chain, dptr,
+				     sizeof(struct pdu_cte_info));
+
+			dptr += sizeof(struct pdu_cte_info);
+			dptr_chain += sizeof(struct pdu_cte_info);
+		}
+
+		/* ADI flag */
+		if (hdr_chain->adi) {
+			(void)memcpy(dptr_chain, dptr,
+				     sizeof(struct pdu_adv_adi));
+
+			dptr += sizeof(struct pdu_adv_adi);
+			dptr_chain += sizeof(struct pdu_adv_adi);
+		}
+
+		/* Finish Common ExtAdv Payload header */
+		com_hdr_chain->adv_mode = 0;
+		com_hdr_chain->ext_hdr_len =
+			dptr_chain - &com_hdr_chain->ext_hdr_adv_data[0];
+
+		/* Prefix overflowed data to chain PDU and reduce the AD data in
+		 * in the current PDU.
+		 */
+		if (ad_len_overflow) {
+			uint8_t *ad_overflow;
+
+			/* Copy overflowed AD data from previous PDU into this
+			 * new chain PDU
+			 */
+			(void)memcpy(&ad_overflow,
+				     &val_ptr[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
+				     sizeof(ad_overflow));
+			ad_overflow += *val_ptr;
+			(void)memcpy(dptr_chain, ad_overflow, ad_len_overflow);
+			dptr_chain += ad_len_overflow;
+
+			/* Reduce the AD data in the previous PDU */
```

Comment:
"in the previous PDU"? Comment in lines 559-561 says about reduction of a AD data in current PDU.